### PR TITLE
Improve image alt support

### LINKED ISSUE #115: alt-text and alt-description specified on images should appear in markup
alt-text and alt-description specified on images in the word doc should should appear in the markup of the page generated for that word doc.

### PR DESCRIPTION
**Pull request by sdmcraft**
Fix #115 

Test URLs:
- Before: https://main--zeiss--hlxsites.hlx.page/drafts/satyam/sample-template
- After: https://issue-115b--zeiss--hlxsites.hlx.page/drafts/satyam/sample-template

P.S. Note the difference in the `img[alt]` attributes between the 2 branches.

### CODE REVIEW

File: helix-query.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ indices:
         select: head > meta[name="description"]
         value: attribute(el, "content")
       imagealt:
-        select: head > meta[property="og:image:alt"]
```
Comment (sdmcraft) [inline helix-query.yaml:21]: [Related slack conversation.](https://cq-dev.slack.com/archives/C9KD0TT6G/p1684322044812229)

File: tools/importer/import.js [hunk 2]
```diff
@@ -159,6 +159,20 @@ const createMetadata = (main, document, url) => {
     meta.Image = el;
   }
 
+  // Preference for image alt tag is og:image:alt > og:title > title
```
Comment (sdmcraft) [inline tools/importer/import.js:162]: Improve importer to import `imagealt` and set it in doc's metadata block. Subsequently this is indexed and gets used in `article-list` block.

File: tools/importer/import.js [hunk 3]
```diff
@@ -159,6 +159,20 @@ const createMetadata = (main, document, url) => {
...
+  let imagealt = document.querySelector('[property="og:image:alt"]');
+  if (imagealt === null || imagealt === undefined) {
```
Comment (jindaliiita) [inline tools/importer/import.js:164]: !imagealt will also work here

File: tools/importer/import.js [hunk 4]
```diff
@@ -159,6 +159,20 @@ const createMetadata = (main, document, url) => {
...
+    const ogTitle = document.querySelector('[property="og:title"]');
+    if (ogTitle) {
+      imagealt = document.querySelector('[property="og:title"]');
```
Comment (jindaliiita) [inline tools/importer/import.js:167]: imagealt = ogtitle || title

File: blocks/carousel/carousel.js [hunk 5]
```diff
@@ -190,13 +190,6 @@ export default async function decorate(block) {
         copyright = copyright.substring(1);
       }
     }
-
-    const img = image.querySelector('img');
```
[the remark this answers is not in the document]
Comment (jindaliiita) [reply blocks/carousel/carousel.js:194]: Is it only present in carousel ?

[the remark this answers is not in the document]
Comment (sdmcraft) [reply blocks/carousel/carousel.js:194]: It's present at other place like the `article-list` but the handling is correct there.

File: tools/importer/import.js (hunk 3 above)
Comment (sdmcraft) [reply tools/importer/import.js:164]: Right `null` and `undefined` are falsey in javascript so your suggestion would also work.

**Review by jindaliiita — APPROVED** [review]

**Pull request merged**